File: src/main/prim/etc/_patchhere.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.core.Syntax
import org.nlogo.agent.{ Patch, Turtle }
import org.nlogo.nvm.{ Context, Reporter }

class _patchhere extends Reporter {
  override def syntax =
    Syntax.reporterSyntax(
      ret = Syntax.PatchType,
      agentClassString = "-T--")
  override def report(context: Context): Patch =
    report_1(context)
  def report_1(context: Context): Patch =
    context.agent.asInstanceOf[Turtle].getPatchHere
}

File: netlogo-gui/src/main/prim/etc/_patchhere.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc;

import org.nlogo.agent.Patch;
import org.nlogo.agent.Turtle;
import org.nlogo.core.Syntax;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;

public final class _patchhere
    extends Reporter {


  @Override
  public Object report(final Context context) {
    return ((Turtle) context.agent).getPatchHere();
  }

  public Patch report_1(final Context context) {
    return ((Turtle) context.agent).getPatchHere();
  }
}
